snapcraft/internal/project_loader/grammar/_processor.py

```diff
@@ -148,7 +171,7 @@ def _parse_dict(self, section: Dict[str, Any], statement: Statement,
                 statement = TryStatement(
                     body=value, processor=self, call_stack=call_stack)
 
-            if _ELSE_CLAUSE_PATTERN.match(key):
+            elif _ELSE_CLAUSE_PATTERN.match(key):
                 _handle_else(statement, value)
```

Comment:
not related, probably a follow-up, but we are missing an else here with a pretty error message, right?

Reply:
It wouldn't hurt, but I wouldn't call it necessarily missing-- the schema would catch such a case.